apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php

```diff
@@ -626,3 +631,28 @@ function wpcom_site_has_global_styles_feature( $blog_id = 0 ) {
 
 	return false;
 }
+
+/**
+ * Checks whether the site has access to the Global Styles feature when the editor is live previewing a Premium theme with a Premium plan or higher.
+ *
+ * @param int $blog_id The WPCOM blog ID.
+ * @return bool Whether the site has access to Global Styles when live previewing.
+ */
+function wpcom_global_styles_is_previewing_premium_theme_with_premium_plan( $blog_id ) {
+	if ( ! isset( $_GET['wp_theme_preview'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		// Not live previewing.
+		return false;
+	}
+	$wp_theme_preview = sanitize_text_field( wp_unslash( $_GET['wp_theme_preview'] ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
+	$is_previewing_premium_theme = str_starts_with( $wp_theme_preview, 'premium/' );
+	if ( ! $is_previewing_premium_theme ) {
+		// Not a premium theme.
+		return false;
+	}
+
+	// Check for a Premium plan or higher by checking if can use global styles.
+	$has_premium_plan_or_higher = wpcom_site_has_feature( WPCOM_Features::GLOBAL_STYLES, $blog_id );
+
+	return $has_premium_plan_or_higher;
```

Comment:
@miksansegundo should this be:
```suggestion
	return ! $has_premium_plan_or_higher;
```

Since we want to return `true` if the user _doesn't_ have premium plan or higher.

Reply:
That logic is tricky, but I think it's correct. 

Try running this in your sandbox for a premium site:

```
// Logs Global Styles feature for your current site
require_once WP_CONTENT_DIR . "/mu-plugins/wpcom-features/functions-wpcom-features.php";
$global_styles_feature = wpcom_site_has_feature( WPCOM_Features::GLOBAL_STYLES ); // The second argument is optional because uses get_current_blog_id() 
l("Global Styles feature for site " . get_current_blog_id() . " is " . ( $global_styles_feature ? 'TRUE' : 'FALSE' ) );
```

Reply:
In other words, our previous tests were false positives. Glad that you suggested to simplify the logic because thanks to that I had the chance to figure out that the previous logic was wrong 😄